packages/ember-metal/lib/is_empty.js

```diff
@@ -39,7 +40,7 @@ export default function isEmpty(obj) {
 
   let objectType = typeof obj;
 
-  if (objectType === 'object') {
+  if (objectType instanceof EmberMap) {
```

Comment:
The fix for this, would be to add some property to `EmberMap` that we can check (e.g. `obj.isEmberMap` or `obj[SOME_PRIVATE_EMBER_MAP_SYMBOL_THINGY]` depending on how paranoid we want to be)...